code/headers/nmea_listener.hpp

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include <usart_connection.hpp>
+
+namespace r2d2::location {
+    class nmea_listener_c {
+    protected:
+        constexpr static uint8_t start_byte = '$';
+        constexpr static uint8_t end_byte = '\n';
+        constexpr static size_t buffer_size = 100;
+
+        usart::usart_connection_c &usart;
+
+        /**
+         * @brief buffer for storing the nmea string
+         *
+         */
+        uint8_t buffer[buffer_size];
+
+        /**
+         * @brief index where the next location of data should be
+         *
+         */
+        size_t index = 0;
+
+        /**
+         * @brief bool for storing if a string is avaialable
+         *
+         */
+        bool available;
+
+    public:
+        nmea_listener_c(usart::usart_connection_c &usart) : usart(usart) {
+        }
+
+        constexpr bool nmea_available() {
+            return available;
+        }
+
+        /**
+         * @brief Get a pointer to a nmea string
+         *
+         * @return const uint8_t*
+         */
+        const uint8_t *get_nmea_string() {
+            if (!nmea_available()) {
+                return nullptr;
+            }
+
+            return buffer;
+        }
+
+        /**
+         * @brief mark the current string as read. To get new data from the
+         * usart.
+         *
+         */
+        constexpr void read_nmea_string() {
```

Comment:
I would rename this function to something like `mark_as_read`, as that is what the function is actually doing.

Reply:
should be fixed in #8